src/modules/formValidationUtils.ts

```diff
@@ -121,3 +121,27 @@ export const validateDepartamento = (valor: string) => {
   const maxLength = valor.length <= MAX_DEPARTAMENTO_FIELD;
   return onlyLetters && maxLength;
 };
+
+export const cleanErrorMessages = (setErrorMessages: any, fieldKey: any) => {
+  setErrorMessages((prevErrors: any) =>
+    prevErrors.filter((error: any) => Object.keys(error)[0] !== fieldKey)
+  );
+};
+
+export const updateErrorMessages = (
+  setErrorMessages: any,
+  fieldKey: any,
+  messageObject: any
+) => {
+  setErrorMessages((prevErrors: any) => {
+    let filteredErrors = prevErrors.filter(
+      (error: any) => Object.keys(error)[0] === fieldKey
+    );
+
+    if (filteredErrors.length === 0) {
+      return [...prevErrors, messageObject];
+    }
+
+    return prevErrors;
+  });
```

Comment:
`updateErrorMessages` é uma função que adiciona erros ao _array_ de erros.
Um erro será incluído nesse _array_ apenas se a chave do objeto de erro for inexistente.
Logo, fazer filtro dos objetos não é algo intuitivo.
Em vez disso, poderia fazer algo como:

```
setErrorMessages((prevErrors: any) => {
    const soughtObject = prevErrors.find((error: any) =>
      error.hasOwnProperty(fieldKey)
    );

    if (soughtObject === undefined) {
      return [...prevErrors, messageObject];
    }

    return [...prevErrors];
  });
```

É apenas uma sugestão. Veja se há outra maneira de fazer isso.

Reply:
Gostei muito dessa alteração, deixa o codigo mais entendivel.